src/pages/Calendar/CalendarContainer.tsx

```diff
@@ -1,69 +1,56 @@
+import { bookingService } from "services";
 import { Calendar } from "./Calendar";
 import { getUTCDateFilter } from "utils/BookingFilters";
 import { AuthContext } from "context/AuthContext";
 import { useInterval } from "hooks/useInterval";
 
-const CalendarContainerComponent = ({ URLRoomId, history }) => {
+const REFRESH_TIME = 5000;
+
+export const CalendarContainer = () => {
+  const { roomId } = useParams();
+  const history = useHistory();
   const [bookings, updateBookings] = useState([]);
   const [allBookings, updateAllBookings] = useState([]);
-  const [roomId, updateRoomId] = useState("");
   const [isLoading, updateIsLoading] = useState(true);
   const [shouldFetch, updateShouldFetch] = useState(false);
   const authContext = useContext(AuthContext);
   const [bookingsHash, updateBookingsHash] = useState("initial");
   const [allBookingsHash, updateAllBookingsHash] = useState("initial");
-  const [delay] = useState(5000); // TODO: when the tab is not used, augment delay time
   const onBookingsDataChange = () => updateShouldFetch(!shouldFetch);
-  const { onLogout } = authContext;
 
   const fetchBookings = async () => {
     try {
-      const reqRoom = await roomService.getOneById(URLRoomId);
       const allData = await bookingService.getAllWithDetails(
         getUTCDateFilter()
       );
 
       const data = await bookingService.getAllWithDetailsByRoom(
         getUTCDateFilter(),
-        URLRoomId
+        roomId
       );
 
-      if (data && allData) {
-        const allBookingsData = allData;
-        const bookingsData = data;
-        // We avoid extra rendering if the data is the same, since this function runs every 1.5 seconds
-        const allBookingsDataStr = JSON.stringify(allBookingsData);
-        const bookingsDataStr = JSON.stringify(bookingsData);
-        if (allBookingsDataStr !== allBookingsHash) {
-          updateAllBookings(allBookingsData);
-          updateAllBookingsHash(allBookingsDataStr);
-        }
+      const allBookingsData = allData;
+      const bookingsData = data;
+      // We avoid extra rendering if the data is the same, since this function runs every 1.5 seconds
+      const allBookingsDataStr = JSON.stringify(allBookingsData);
+      const bookingsDataStr = JSON.stringify(bookingsData);
 
-        if (bookingsDataStr !== bookingsHash) {
-          updateBookings(bookingsData);
-          updateBookingsHash(bookingsDataStr);
-        }
-      } else {
-        const { name } = data;
-        // The webtoken is invalid for any reason
-        if (name === "JsonWebTokenError") {
-          onLogout();
-          return history.push("/login");
-        }
+      if (allBookingsDataStr !== allBookingsHash) {
+        updateAllBookings(allBookingsData);
+        updateAllBookingsHash(allBookingsDataStr);
       }
 
-      if (typeof reqRoom === "object") {
-        updateIsLoading(false);
-        return updateRoomId(URLRoomId);
+      if (bookingsDataStr !== bookingsHash) {
```

Comment:
I was thinking that we could just avoid doing this "hash" comparisons, just using `useMemo`

Reply:
I was thinking the same thing, but i don't have any experience with useMemo. i'll just remove the hashing, maybe in the future we can use it.